Marlin/ultralcd.cpp

```diff
@@ -3752,9 +3752,23 @@ void kill_screen(const char* lcd_msg) {
      * "Print from SD" submenu
      *
      */
-    void lcd_sdcard_menu() {
+    #if ENABLED(SD_REPRINT_LAST_SELECTED_FILE)
+      uint32_t saved_encoderPosition = 0;
+      static millis_t last_sdcard_time = 0;
+    #endif
+
+void lcd_sdcard_menu() {
       ENCODER_DIRECTION_MENUS();
-      if (!lcdDrawUpdate && !lcd_clicked) return; // nothing to do (so don't thrash the SD card)
+  
+      #if ENABLED(SD_REPRINT_LAST_SELECTED_FILE)
+        const millis_t ms = millis();
+        if (last_sdcard_time+5000 < ms) {            // if the printer has been busy printing, lcd_sdcard_menu() should not 
```

Comment:
For comparing time to `millis()` it is absolutely required to use the `ELAPSED` or `PENDING` macros. It will also be a little faster if you set a `next_sdcard_time` instead as `ms + 5000` rather than including the add in every test.